components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -120,6 +120,25 @@ class GeckoEngine(
         }
     }
 
+    /**
+     * Add support for service workers requests to open new tabs.
+     *
+     * @param onAddTab - callback to adding a tab using the provided [EngineSession].
+     * Implementations are expected to add a new normal (not private) tab but not load any URL in that.
+     * The background service worker will use the same passed in [EngineSession] to load it's needed URL.
+     */
+    fun installServiceWorkersSupport(
+        onAddTab: (engineSession: EngineSession) -> Unit
+    ) {
+        runtime.serviceWorkerDelegate = GeckoRuntime.ServiceWorkerDelegate {
+            val newEngineSession = GeckoEngineSession(runtime, false, defaultSettings, openGeckoSession = false)
```

Comment:
`openGeckoSession = false` in preparation of https://bugzilla.mozilla.org/show_bug.cgi?id=1759588